### PR TITLE
Update conv.jl

### PR DESCRIPTION
**Pull request by vivekkumar7089**
Specifying the format of the data given to the convolutional layer, as mentioned in #1465 .

### CODE REVIEW

**Comment by DhairyaLGandhi** [comment]
LGTM

bors r+

File: src/layers/conv.jl [hunk 1]
```diff
@@ -40,7 +40,8 @@ Image data should be stored in WHCN order (width, height, channels, batch).
 In other words, a 100×100 RGB image would be a `100×100×3×1` array,
 and a batch of 50 would be a `100×100×3×50` array.
 This has `N=2` spatial dimensions, and needs a kernel size like `(5,5)`,
-a 2-tuple of integers.
+a 2-tuple of integers. For different `N`, data should be `N_1 x N_2 x ... x N_n x channels x batch`,
+i.e. `spatial/temporal dimensions x channels x batch`.
```
Comment (DhairyaLGandhi) [inline src/layers/conv.jl:44]:
```suggestion
i.e. `spatial/temporal dimensions x channels x batch` (notice that the batch is the last dimension).
```

File: src/layers/conv.jl [hunk 2]
```diff
@@ -40,7 +40,8 @@ Image data should be stored in WHCN order (width, height, channels, batch).
 In other words, a 100×100 RGB image would be a `100×100×3×1` array,
 and a batch of 50 would be a `100×100×3×50` array.
 This has `N=2` spatial dimensions, and needs a kernel size like `(5,5)`,
-a 2-tuple of integers.
+a 2-tuple of integers. For different `N`, data should be `N_1 x N_2 x ... x N_n x channels x batch`,
```
Comment (ToucheSir) [inline src/layers/conv.jl:43]: Thanks for the PR! A couple of comments:

1. "2-tuple" is obviously no longer correct when we're talking about convolutions with arbitrary dimensions.
2. I used `monospace` text in the original comment for convenience, but the dimensions ought to be properly latexified here. See https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html#torch.nn.Conv2d for reference.
3. Likewise, I wrote "spatial/temporal dimensions" as a shorthand that everyone in the issue thread would understand, but that language should be made more precise and elaborated upon for a docstring. For example, a video may have 2 spatial dimensions + 1 temporal dimension, whereas a 3d volume has 3 spatial dimensions. Ideally neither term should be required to explain the expect format of the kernel.

Comment (DhairyaLGandhi) [reply src/layers/conv.jl:43]: I think describing it too specifically might hurt it.  Giving examples is fine, and I think the video example is good, but important to note that the examples can go beyond images or whatever and be generalised to regular feature arrays.

Comment (ToucheSir) [reply src/layers/conv.jl:43]: Agreed. I should clarify that these examples are meant to help _write_ the docstring, and should not appear in the docstring itself. That should aim to be as domain agnostic as possible, hence my point about not using "spatial" or "temporal" unless needed.

Comment (vivekkumar7089) [reply src/layers/conv.jl:43]: I think, we can give another example for 1D like already given for 2D (WHCN), and we will also mention that channels and batch size should be the 2nd last and last arguments, respectively. Is it fine @ToucheSir @DhairyaLGandhi ?

**Pull request closed without merging**